include/boost/math/quadrature/gauss_kronrod.hpp

```diff
@@ -45,17 +45,17 @@ class gauss_kronrod_detail
    {
       std::vector<Real> result(abscissa().size(), 0);
       unsigned gauss_order = (N - 1) / 2;
-      unsigned gauss_start = gauss_order & 1 ? 0 : 1;
+      unsigned gauss_start = (gauss_order & 1) ^ 1;
       const legendre_stieltjes<Real>& E = get_legendre_stieltjes();
 
-      for (unsigned i = gauss_start; i < abscissa().size(); i += 2)
+      for (size_t i = gauss_start; i < abscissa().size(); i += 2)
       {
          Real x = abscissa()[i];
          Real p = boost::math::legendre_p_prime(gauss_order, x);
          Real gauss_weight = 2 / ((1 - x * x) * p * p);
          result[i] = gauss_weight + static_cast<Real>(2) / (static_cast<Real>(gauss_order + 1) * legendre_p_prime(gauss_order, x) * E(x));
       }
-      for (unsigned i = gauss_start ? 0 : 1; i < abscissa().size(); i += 2)
+      for (size_t i = gauss_start ^ 1; i < abscissa().size(); i += 2)
```

Comment:
Surely this is simply not the same thing?

Reply:
Yeah it is.

Reply:
guass_start is 0 or 1. Though, I wonder if it may be optimized to the same thing anyway.